src/Compat.jl

```diff
@@ -51,6 +51,15 @@ end
     end
 end
 
+# https://github.com/JuliaLang/julia/pull/43852
+if isdefined(Base, Symbol("@assume_effects"))
```

Comment:
```suggestion
if VERSION < v"1.8.0-DEV.1484"
```
Makes it easier in the future to figure out whether we can remove this when we bump the minimum Julia version.

Reply:
Whoops, that reverses the logic though, sorry about that.